### PR TITLE
Update cfg of balloon_in_use

### PR DESCRIPTION
**Pull request by suqinhuang**
Signed-off-by: Suqin Huang <shuang@redhat.com>

id: 1416288

### CODE REVIEW

**Comment by suqinhuang** [comment]
PASS 01-smp_16.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.balloon_in_use.with_stop_continue.during_bg_test

PASS 02-smp_16.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.balloon_in_use.with_shutdown.during_bg_test

PASS
03-smp_16.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.balloon_in_use.with_reboot.during_bg_test

PASS 04-smp_16.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.balloon_in_use.with_system_reset.during_bg_test

**Comment by kimi1978wy** [comment]
@shuang,
 test on win2016/win8.1-32, still fail as "TestFail: Backgroud test balloon_stress is not alive!" on my side. 
 (1/1) smp_8.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win8.i386.1.io-github-autotest-qemu.balloon_in_use.with_stop_continue.during_bg_test: FAIL (190.12 s)
cml:
python ConfigTest.py --testcase=balloon_in_use.with_stop_continue.during_bg_test --guestname=Win8..1 --platform=i386 --clone=no

**Comment by suqinhuang** [comment]
https://github.com/autotest/tp-qemu/pull/924

**Comment by suqinhuang** [comment]
@kimi1978wy could you try with this patch applied https://github.com/autotest/tp-qemu/pull/924

**Comment by kimi1978wy** [comment]
@suqinhuang tried with #924 , failed w/ 16G mem, but pass w/ 3G mem.
 smp_8.**3072m**.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.balloon_in_use.with_stop_continue.during_bg_test: **PASS** (430.64 s)

smp_8.**16384m**.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.balloon_in_use.with_stop_continue.during_bg_test: **FAIL** (794.66 s)

Failed as 
00:24:05 DEBUG| TestFail: Failed to balloon memory to expect value during 100.0s    [context: Change VM memory to 15056]
00:24:05 DEBUG| 
00:24:55 WARNI| No enough free memory to launch VMs, reset guest memory to 6400 MB

detailed debug log refer to:
[stop-cont.zip](https://github.com/autotest/tp-qemu/files/762724/stop-cont.zip)

**Comment by suqinhuang** [comment]
this patch fix the maxmem issue
https://github.com/avocado-framework/avocado-vt/pull/878
@kimi1978wy

**Comment by suqinhuang** [comment]
@kimi1978wy update balloon_timeout to 240s, could you have a try, it pass in my host.

Thanks

**Comment by suqinhuang** [comment]
PASS 1-smp_8.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.balloon_in_use.with_stop_continue.during_bg_test

**Comment by kimi1978wy** [comment]
@suqinhuang 
tried on win8.1-64
1  the "driver_load_stress.with_balloon.during_bg_test" is still fail, please help to check. error still "TestFail: Backgroud test balloon_stress is not alive!" log attached

2 in "balloon_in_use.with_live_migration.during_bg_test", the migration is occurred after the video stopped. log attached.

 (1/6) smp_8.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.driver_load_stress.with_balloon.during_bg_test: FAIL (197.06 s)
 (2/6) smp_8.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.balloon_in_use.with_stop_continue.during_bg_test: PASS (341.53 s)
 (3/6) smp_8.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.balloon_in_use.with_shutdown.during_bg_test: PASS (543.78 s)
 (4/6) smp_8.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.balloon_in_use.with_reboot.during_bg_test: PASS (367.22 s)
 (5/6) smp_8.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.balloon_in_use.with_system_reset.during_bg_test: PASS (437.16 s)
 (6/6) smp_8.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.balloon_in_use.with_live_migration.during_bg_test: PASS (1383.13 s)

**Comment by kimi1978wy** [comment]
@suqinhuang log for driver_load_stress and with_live_migration
[debug-load.log.zip](https://github.com/autotest/tp-qemu/files/792412/debug-load.log.zip)
[debug-migrate.log.zip](https://github.com/autotest/tp-qemu/files/792413/debug-migrate.log.zip)

**Comment by kimi1978wy** [comment]
background test balloon_stress is alive now after applying this PR.
cml:python ConfigTest.py --testcase=balloon_in_use.with_stop_continue.during_bg_test,balloon_in_use.with_shutdown.during_bg_test,balloon_in_use.with_reboot.during_bg_test,balloon_in_use.with_system_reset.during_bg_test,balloon_in_use.with_live_migration.during_bg_test --guestname=Win8..1 --platform=x86_64 --clone=no

results:
 (1/5) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.balloon_in_use.with_stop_continue.during_bg_test: PASS (531.43 s)
 (2/5) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.balloon_in_use.with_shutdown.during_bg_test: PASS (406.14 s)
 (3/5) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.balloon_in_use.with_reboot.during_bg_test: PASS (464.55 s)
 (4/5) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.balloon_in_use.with_system_reset.during_bg_test: PASS (488.93 s)
 (5/5) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.balloon_in_use.with_live_migration.during_bg_test: PASS (1375.60 s)

"driver_load_stress.with_balloon.during_bg_test" tracked with https://bugzilla.redhat.com/show_bug.cgi?id=1426039

 Acked-by: Yu Wang wyu@redhat.com

File: qemu/tests/cfg/balloon_in_use.cfg [hunk 1]
```diff
@@ -7,32 +7,35 @@
...
+    #Disable first startup guide for windows media player
+    wmplayer_reg_cmd = "reg add HKLM\SOFTWARE\Policies\Microsoft\WindowsMediaPlayer /v GroupPrivacyAcceptance  /t REG_DWORD /f /d 00000001"
+    wmplayer_path = "${program_files}\Windows Media Player\wmplayer.exe"
+    #Install kmplayer if wmplayer is not installed default
+    kmplayer_install_cmd = "start /wait WIN_UTILS:\kmplayer\%s\KMPlayer-setup.exe /SP- /VERYSILENT"
+    kmplayer_path = "${program_files}\KMPlayer\kmplayer.exe"
+    video_url = http://fileshare.com/Peppa_Pig_39_The_Tree_House.avi
+    play_video_cmd = '"%s" "%s" /play /fullscreen'
+    target_process = mplayer
+    guest_alias = "Win2008-sp2-32:2k8\x86,Win2008-sp2-64:2k8\amd64,Win2008-r2-64:2k8\amd64,Win2012-64:2k12\amd64,Win2012-64r2:2k12\amd64"
```
Comment (CongLi) [inline qemu/tests/cfg/balloon_in_use.cfg]: how about win10 and win2016

Comment (suqinhuang) [inline qemu/tests/cfg/balloon_in_use.cfg]: wmplayer is installed in win10 and win2016

**Comment by suqinhuang** [comment]
@pingl2016 could you help review this patch

File: qemu/tests/cfg/balloon_in_use.cfg [hunk 2]
```diff
@@ -7,32 +7,35 @@
...
     run_bgstress = balloon_stress
+    video_test = win_video_play
     bg_stress_run_flag = balloon_test
     set_bg_stress_flag = yes
     free_mem_cmd = wmic os get FreePhysicalMemory
     default_memory = ${mem}
-    video_url = http://FILESHARE.COM/pub/section2/kvmauto/video/big_buck_bunny_480p_stereo.avi
+    session_cmd_timeout = 240
+    balloon_timeout = 240
+    need_reboot = yes
```
Comment (CongLi) [inline qemu/tests/cfg/balloon_in_use.cfg]: where is this param used?

**Comment by suqinhuang** [comment]
PASS 01-smp_4.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.balloon_in_use.with_stop_continue.before_bg_test
PASS 02-smp_4.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.balloon_in_use.with_stop_continue.during_bg_test
PASS 03-smp_4.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.balloon_in_use.with_stop_continue.after_bg_test
PASS 04-smp_4.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.balloon_in_use.with_shutdown.before_bg_test
PASS 05-smp_4.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.balloon_in_use.with_shutdown.during_bg_test
PASS 06-smp_4.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.balloon_in_use.with_shutdown.after_bg_test
PASS 07-smp_4.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.balloon_in_use.with_reboot.before_bg_test
PASS 08-smp_4.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.balloon_in_use.with_reboot.during_bg_test
PASS 09-smp_4.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.balloon_in_use.with_reboot.after_bg_test

**Comment by suqinhuang** [comment]
@kimi1978wy 924 is merged, and this patch is updated, could you review it.

Thanks

**Comment by kimi1978wy** [comment]
@suqinhuang  phou is in charge of this patch now, she will review it and comment on this.

Thanks
Yu Wang

**Comment by peixiu** [comment]
Test these job after applying this patch.
cml:python ConfigTest.py --testcase=balloon_in_use.with_stop_continue.during_bg_test,balloon_in_use.with_shutdown.during_bg_test,balloon_in_use.with_reboot.during_bg_test,balloon_in_use.with_system_reset.during_bg_test,balloon_in_use.with_live_migration.during_bg_test --guestname=Win8..1 --platform=x86_64 --clone=no
(1/6) smp_16.8192m.repeat1.pre_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_native: PASS (2297.83 s)
 (2/6) smp_16.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.balloon_in_use.with_stop_continue.during_bg_test: PASS (378.06 s)
 (3/6) smp_16.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.balloon_in_use.with_shutdown.during_bg_test: PASS (268.38 s)
 (4/6) smp_16.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.balloon_in_use.with_reboot.during_bg_test: PASS (320.32 s)
 (5/6) smp_16.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.balloon_in_use.with_system_reset.during_bg_test: PASS (333.90 s)

Acked-by: Peixiu Hou phou@redhat.com

**Comment by suqinhuang** [comment]
@peixiu patch is updated, could you have a loop

**Comment by suqinhuang** [comment]
@peixiu pls ignore migration sub test, as the design is updated, it will be handled in https://github.com/autotest/tp-qemu/pull/1024

**Comment by peixiu** [comment]
@suqinhuang ok, thanks~
After applying this PR, follows job are work normally:
balloon_in_use.with_stop_continue.during_bg_test
balloon_in_use.with_shutdown.during_bg_test
balloon_in_use.with_reboot.during_bg_test
balloon_in_use.with_system_reset.during_bg_test

And for balloon_in_use.with_live_migration.during_bg_test, will track through #1024

Acked-by: Peixiu Hou phou@redhat.com

**Comment by suqinhuang** [comment]
@luckyh hi, could you have a look this patch

**Pull request merged**